bridge-token-factory/src/lib.rs

```diff
@@ -372,6 +582,10 @@ impl BridgeTokenFactory {
             target, message
         ));
 
+        let amount_to_transfer: u128;
+        let fee_amount = self.calculate_deposit_fee_amount(&token, amount, &message);
```

Comment:
fee amount calculated incorrectly: you should take percent from amount_to_transfer, not from amount.

Reply:
The `amount_to_transfer` = `amount ` - `fee_amount `

Reply:
See comment here: https://github.com/aurora-is-near/rainbow-token-connector/pull/213#discussion_r1264986513